web/ui/mantine-ui/src/state/settingsSlice.ts

```diff
@@ -30,6 +30,32 @@ export const localStorageKeyEnableSyntaxHighlighting =
 export const localStorageKeyEnableLinter = "settings.enableLinter";
 export const localStorageKeyShowAnnotations = "settings.showAnnotations";
 
+// This dynamically/generically determines the pathPrefix by stripping the first known
+// endpoint suffix from the window location path. It works out of the box for both direct
+// hosting and reverse proxy deployments with no additional configurations required.
+const getPathPrefix = (path: string) => {
+  if (path.endsWith("/")) {
+    path = path.slice(0, -1);
+  }
+
+  const pagePaths = [
+    "/query",
+    "/alerts",
+    "/targets",
+    "/rules",
+    "/service-discovery",
+    "/status",
+    "/tsdb-status",
+    "/flags",
+    "/config",
+    "/alertmanager-discovery",
+    "/agent",
+  ];
+
+  const pagePath = pagePaths.find((p) => path.endsWith(p));
+  return path.slice(0, path.length - (pagePath || "").length);
```

Comment:
as an alternative:

```typescript
  for (const path of paths) {
    if (path && basePath.endsWith(path)) {
      basePath = basePath.slice(0, basePath.length - path.length);
      break;
    }
  }
```

Will just avoid to write `pagePath || ""`. But it's just matter of taste. Feel free to ignore it :)

Reply:
Thanks! Will ignore for now since it introduces extra changes that could go wrong, but feel free to improve later :)